src/app/layout/spinner/spinner.service.ts

```diff
@@ -1,18 +1,40 @@
 import { Injectable } from '@angular/core';
 import { select, Store } from '@ngrx/store';
-import { State } from '@store/.';
-import { spinnerState, SpinnerState } from '@store/spinner/reducers/spinner.reducer';
-import { Subject, combineLatest, map, Observable } from 'rxjs';
+import { spinnerState } from '@store/spinner/reducers/spinner.reducer';
+import { technicalRecordsLoadingState } from '@store/technical-records';
+import { testResultLoadingState } from '@store/test-records';
+import { combineLatest, map, Observable, withLatestFrom } from 'rxjs';
 
 @Injectable({
   providedIn: 'root'
 })
 export class SpinnerService {
+  globalLoadingState$: Observable<boolean>;
+  testResultLoadingState$: Observable<boolean>;
+  techRecordsLoadingState$: Observable<boolean>;
 
   constructor(private store: Store) {
+    this.globalLoadingState$ = this.store.pipe(select(spinnerState));
```

Comment:
I like the moving of loading state back into the reducers that are triggering them, but what is changing the globalLoadingState?

Reply:
I added an action we can use to change global state but at the moment nothing is using that as we don't have any non API process that might need a spinner. Later if we have any reason to show the spinner that doesn't involve an API we can use that action.